### PR TITLE
Do not claim disk that doesn't belong to Ceph.

### PR DESCRIPTION
**Pull request by jsuchome**
When there is no disk with Ceph role, take disks with no role.

Originally https://github.com/crowbar/barclamp-ceph/pull/168

Requires https://github.com/crowbar/crowbar-core/pull/161

### CODE REVIEW

**Comment by rsalevsky** [comment]
+1 after fixing the hound issues

File: chef/cookbooks/ceph/recipes/osd.rb [hunk 1]
```diff
@@ -69,7 +69,11 @@
   if is_crowbar?
     node.set["ceph"]["osd_devices"] = [] if node["ceph"]["osd_devices"].nil?
     min_size_blocks = node["ceph"]["osd"]["min_size_gb"] * 1024 * 1024 * 2
-    unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node).sort.select { |d| d.size >= min_size_blocks }
+    claim_string = "Ceph"
+    # claim only disks with Ceph role; when there is none, take disks without role
+    ceph_disks, disks_without_role = BarclampLibrary::Barclamp::Inventory::Disk.prepared_for(node, claim_string)
```
Comment (jsuchome) [inline chef/cookbooks/ceph/recipes/osd.rb:74]: I'd rather leave this one as it is

Comment (tboerger) [inline chef/cookbooks/ceph/recipes/osd.rb:74]: Even that can be resolved with

```
ceph_disks, disks_without_role = BarclampLibrary::Barclamp::Inventory::Disk.prepared_for(
  node,
  claim_string
)
```

**Comment by tboerger** [comment]
:+1:

File: chef/cookbooks/ceph/recipes/osd.rb (hunk 1 above)
Comment (jsuchome) [inline chef/cookbooks/ceph/recipes/osd.rb:74]: I know it could, I just like the one line way :-)

Comment (mjura) [inline chef/cookbooks/ceph/recipes/osd.rb:74]: jsuchome: We have to train to be immortal `line breakers`

**Comment by jsuchome** [comment]
crowbar-ceph not used any more

**Pull request closed without merging**